iree/tools/init_passes.h

```diff
@@ -55,83 +55,30 @@ namespace mlir {
-      []() -> std::unique_ptr<Pass> { return mlir::createCSEPass(); });
-  ::mlir::registerPass("canonicalize", "Canonicalize operations",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createCanonicalizerPass();
-                       });
-  ::mlir::registerPass(
-      "inline", "Inline function calls",
-      []() -> std::unique_ptr<Pass> { return mlir::createInlinerPass(); });
-  ::mlir::registerPass("snapshot-op-locations",
-                       "Generate new locations from the current IR",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createLocationSnapshotPass();
-                       });
-  ::mlir::registerPass(
-      "loop-coalescing",
-      "Coalesce nested loops with independent bounds into a single loop",
-      []() -> std::unique_ptr<Pass> {
-        return mlir::createLoopCoalescingPass();
-      });
-  ::mlir::registerPass("loop-invariant-code-motion",
-                       "Hoist loop invariant instructions outside of the loop",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createLoopInvariantCodeMotionPass();
-                       });
-  ::mlir::registerPass("memref-dataflow-opt",
-                       "Perform store/load forwarding for memrefs",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createMemRefDataFlowOptPass();
-                       });
-  ::mlir::registerPass(
-      "parallel-loop-collapsing",
-      "Collapse parallel loops to use less induction variables",
-      []() -> std::unique_ptr<Pass> {
-        return mlir::createParallelLoopCollapsingPass();
-      });
-  ::mlir::registerPass(
-      "print-op-stats", "Print statistics of operations",
-      []() -> std::unique_ptr<Pass> { return mlir::createPrintOpStatsPass(); });
-  ::mlir::registerPass("strip-debuginfo",
-                       "Strip debug info from all operations",
-                       []() -> std::unique_ptr<Pass> {
-                         return mlir::createStripDebugInfoPass();
-                       });
-  ::mlir::registerPass(
-      "symbol-dce", "Eliminate dead symbols",
-      []() -> std::unique_ptr<Pass> { return mlir::createSymbolDCEPass(); });
-  createCanonicalizerPass();
-  createCSEPass();
+  // Init general passes
+#define GEN_PASS_REGISTRATION_AffineLoopFusion
+#define GEN_PASS_REGISTRATION_AffinePipelineDataTransfer
+#define GEN_PASS_REGISTRATION_CSE
+#define GEN_PASS_REGISTRATION_Canonicalizer
+#define GEN_PASS_REGISTRATION_Inliner
+#define GEN_PASS_REGISTRATION_LocationSnapshot
+#define GEN_PASS_REGISTRATION_LoopCoalescing
+#define GEN_PASS_REGISTRATION_LoopInvariantCodeMotion
+#define GEN_PASS_REGISTRATION_MemRefDataFlowOpt
+#define GEN_PASS_REGISTRATION_ParallelLoopCollapsing
+#define GEN_PASS_REGISTRATION_PrintOpStats
+#define GEN_PASS_REGISTRATION_StripDebugInfo
+#define GEN_PASS_REGISTRATION_SymbolDCE
```

Comment:
Please undef all of these after :-/

Which brings me to my next point. Is this actually better? I think I'd maybe like to see better infra from core before we adopt this

Reply:
These are undef after using there were *used* in `mlir/Transforms/Passes.h.inc`.

That would be the other option. At the moment we the code is just inlined to `init_passes.h`. Don't know what is better.